File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sumOfLeftLeaves(TreeNode root) {
        return dfs(root, false);
    }
    public int dfs(TreeNode node, boolean bool){
        if (node == null) return 0;
        if (node.left == null && node.right == null){
            return bool ? node.val : 0;
        }
        
        int left = dfs(node.left, true);
        int right = dfs(node.right, false);
        return left+right;
    }
}

File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        def dfs(node, val):
            if not node:
                return 0
            if not node.left and not node.right:
                if val:
                    return node.val
                else:
                    return 0
                
            left = dfs(node.left, True)
            right = dfs(node.right, False)
            return left + right
        
        return dfs(root, False)